front/src/common/IntervalsDataViz/dataviz.tsx

```diff
@@ -92,15 +122,80 @@ export const LinearMetadataDataviz = <T extends { [key: string]: any }>({
   }, [data, field]);
 
   /**
-   * When data or viewbox change
+   * When data change
    * => we recompute the data for the viz
    * => we recompute the full length of the displayed data.
+   * => we recompute the objects that are absolutely positioned
    */
   useEffect(() => {
```

Comment:
I am not a fan of this pattern:
setMustRedraw(true) -> do ops & redraw -> setMustRedraw(false)
It is error-creating and difficult to maintain
Do your ops on single useEffect and do not provoke redraw cascade please :)

Reply:
Done, I fixed it !